pkg/metrics/openai_recorder.go

```diff
@@ -197,13 +228,31 @@ func (r *OpenAIRecorder) RecordResponse(id, model string, rw http.ResponseWriter
 	}
 }
 
-func (r *OpenAIRecorder) convertStreamingResponse(streamingBody string) string {
+func (r *OpenAIRecorder) convertStreamingResponse(streamingBody string) (string, int) {
 	lines := strings.Split(streamingBody, "\n")
 	var contentBuilder strings.Builder
 	var reasoningContentBuilder strings.Builder
 	var lastChoice, lastChunk map[string]interface{}
+	errorStatusCode := 0 // 0 means no error detected
 
 	for _, line := range lines {
+		// Check for error lines in streaming format
+		if strings.HasPrefix(line, "error: ") {
+			errorData := strings.TrimPrefix(line, "error: ")
+			var errorObj map[string]interface{}
+			if err := json.Unmarshal([]byte(errorData), &errorObj); err == nil {
+				// Extract error code if available
+				if code, ok := errorObj["code"].(float64); ok {
+					errorStatusCode = int(code)
```

Comment:
Do you know that this can't be 400?
Why don't we return an error instead of an int to signal an error?

Reply:
@doringeman  addressed in [95c2c78](https://github.com/docker/model-runner/pull/160/commits/95c2c789e5825ea5b95cfd3d3fd5468de560e69b)